### PR TITLE
wasi policy

### PR DESCRIPTION
**Pull request by flavio**
> Note: This policy is based on https://github.com/kubewarden/docs/pull/194

Explain how and when to write a plain WASI policy

TODO: create a template repository that contains a Go-based WASI policy and link it inside of the docs

### CODE REVIEW

File: docs/writing-policies/wasi.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+Each Kubewarden policy must be annotated via the `kwctl annotate` command.
+The policy metadata of a plain WASI policy must provide this value:
+
+```yaml
+executionMode: wasi
+```
+
+## Template project
+
+TODO: create a Go-based template project
```
Comment (jvanz) [inline docs/writing-policies/wasi.md:133]:
```suggestion
:::note
Go-based template project is yet to be created.
:::

```

Comment (jvanz) [reply docs/writing-policies/wasi.md:133]: https://github.com/kubewarden/kubewarden-controller/issues/472

**Comment by flavio** [comment]
Moving to blocked, waiting for https://github.com/kubewarden/kubewarden-controller/issues/472 to be closed so that we can put the link to it inside of the docs

File: docs/writing-policies/wasi.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+more.
+
+Actually, many of the compilers used to compile Kubewarden policies
+produce WebAssembly modules that targets the WASI interfaces.
+However, Kubewarden policies leverage the [waPC](https://github.com/wapc)
+project to implement a bi-directional communication between the
+policy and the policy runtime (`kwctl` or `policy-server`).
+
+There are however some special cases when the waPC project cannot be
+used. In these limited circumstances it's possible to write a policy
```
Comment (viccuad) [inline docs/writing-policies/wasi.md:30]:
```suggestion
used yet. In these limited circumstances it's possible to write a policy
```

**Review by viccuad — APPROVED** [review]
LGTM!

**Comment by flavio** [comment]
Moving to blocked, this has to be merged once Kubewarden 1.7.0 is out

**Comment by jvanz** [comment]
Merging. `v1.7.0-rc1` is out

**Pull request merged**